feat(collection): add concat (#748)
instill-ai/pipeline-backend: pkg/component/generic/collection/v0/.compogen/bottom.mdx

> 

Reviewer: As mentioned in another PR, instead of adding newlines we should put that in the `compogen` template. This is a workaround and contributors (myself included) don't need to know about this.

Author: I remove `-` in compogen when there is extra content

## Example Recipes

> In this example, we are combining two arrays of strings using the `union` and `concat` components.

Reviewer: nit

```suggestion
### Combine 2 arrays using `TASK_UNION` and `TASK_CONCAT`

```

```yaml

> variable:

Reviewer: How about documenting and providing examples with a comment here? E.g.

```yaml
# This pipeline combines 2 arrays of strings
# Examples:
# ["foo", "bar"], ["zot", "bat"] -> ["foo", "bar", "zot", "bat"], []

Author: I added these 2 in corresponding components.
```
# This component concatenates two arrays of strings.
# Examples:
# ["foo", "bar"], ["zot", "bat"] -> ["foo", "bar", "zot", "bat"]
```

```
# This component combines two arrays of strings in a union, ensuring that no duplicates are included.
# Examples:
# ["foo", "bar"], ["foo", "bat"] -> ["foo", "bar", "bat"]
```

Reviewer: I meant documented the input/output of the **pipelines**, not the individual components. But it's fine.

  texts:
    instill-format: array:string
    title: Text
  texts-2:
    instill-format: array:string
    title: Text

component:
  union:
    type: collection
    input:
      sets:
        - ${variable.texts}
        - ${variable.texts-2}
    condition:
    task: TASK_UNION
  concat:
    type: collection
    input:
      arrays:
        - ${variable.texts}
        - ${variable.texts-2}
    condition:
    task: TASK_CONCAT

output:
  union-result:
    title: Union Result
    value: ${union.output.set}
  concat-result:
    title: Concat Result
    value: ${concat.output.array}
```

> In this example, we are creating an object with keys `type` and `text` from a string.

Reviewer: nit
```suggestion
### Use `TASK_ASSIGN` to create an object with `type` and `text` keys

```
